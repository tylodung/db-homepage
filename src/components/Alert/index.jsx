import React from "react"
import PropTypes from "prop-types"
import Icon from "@fortawesome/react-fontawesome"

import styles from "./index.module.css"

const Alert = ({ message, description, type }) => (
  <div className={`${styles.root} ${styles[type]}`} role="alert">
    <h3><Icon icon={type === `error` ? `exclamation-circle` : `check-circle`} fixedWidth /> {message}</h3>
    <p>{description}</p>
  </div>
)

Alert.propTypes = {
  type: PropTypes.string.isRequired,
  message: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
}

export default Alert
