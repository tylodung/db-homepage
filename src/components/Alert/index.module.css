.root {
  color: black;
  padding: 10px 20px;
  margin-bottom: 1.25rem; /* 20/16 */
}

.root.error {
  background: #ffc6c3;
}

.root.success {
  background: #b9ffcc;
}
