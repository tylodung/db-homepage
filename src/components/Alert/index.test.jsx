import React from "react"
import { shallow } from "enzyme"
import Alert from "./"

describe(`Alert`, () => {
  let props
  let mountedComponent
  const wrapper = () => {
    if (!mountedComponent) {
      mountedComponent = shallow(<Alert {...props} />)
    }
    return mountedComponent
  }

  beforeEach(() => {
    props = {
      type: `error`,
      message: `Alert error title`,
      description: `This is the alert body text`,
    }
    mountedComponent = undefined
  })

  it(`renders an Alert correctly`, () => {
    expect(wrapper()).toMatchSnapshot()
  })
})
