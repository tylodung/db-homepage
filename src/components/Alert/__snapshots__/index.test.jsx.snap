// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert renders an Alert correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Alert
    description="This is the alert body text"
    message="Alert error title"
    type="error"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          <FontAwesomeIcon$1
            border={false}
            className=""
            compose={null}
            fixedWidth={true}
            flip={null}
            icon="exclamation-circle"
            listItem={false}
            name=""
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            symbol={false}
            transform={null}
          />
           
          Alert error title
        </h3>,
        <p>
          This is the alert body text
        </p>,
      ],
      "className": "root error",
      "role": "alert",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <FontAwesomeIcon$1
              border={false}
              className=""
              compose={null}
              fixedWidth={true}
              flip={null}
              icon="exclamation-circle"
              listItem={false}
              name=""
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              transform={null}
            />,
            " ",
            "Alert error title",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "border": false,
              "className": "",
              "compose": null,
              "fixedWidth": true,
              "flip": null,
              "icon": "exclamation-circle",
              "listItem": false,
              "name": "",
              "pull": null,
              "pulse": false,
              "rotation": null,
              "size": null,
              "spin": false,
              "symbol": false,
              "transform": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          " ",
          "Alert error title",
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "This is the alert body text",
        },
        "ref": null,
        "rendered": "This is the alert body text",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            <FontAwesomeIcon$1
              border={false}
              className=""
              compose={null}
              fixedWidth={true}
              flip={null}
              icon="exclamation-circle"
              listItem={false}
              name=""
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              transform={null}
            />
             
            Alert error title
          </h3>,
          <p>
            This is the alert body text
          </p>,
        ],
        "className": "root error",
        "role": "alert",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <FontAwesomeIcon$1
                border={false}
                className=""
                compose={null}
                fixedWidth={true}
                flip={null}
                icon="exclamation-circle"
                listItem={false}
                name=""
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                transform={null}
              />,
              " ",
              "Alert error title",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "border": false,
                "className": "",
                "compose": null,
                "fixedWidth": true,
                "flip": null,
                "icon": "exclamation-circle",
                "listItem": false,
                "name": "",
                "pull": null,
                "pulse": false,
                "rotation": null,
                "size": null,
                "spin": false,
                "symbol": false,
                "transform": null,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            " ",
            "Alert error title",
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "This is the alert body text",
          },
          "ref": null,
          "rendered": "This is the alert body text",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
